### PR TITLE
SR v4 Servo Board

### LINKED ISSUE #59: Support SR v4 Power Board
Support SR v4 Power Board.

Prerequisites:
- ~~Button Support~~ - #44 
- ~~Power Output Support~~ - #61
- ~~Battery (voltage / status)~~ - #62
- ~~Piezo~~ - #60

### PR DESCRIPTION
**Pull request by trickeydan**
Resolves #59

### CODE REVIEW

File: j5/boards/sr/v4/servo_board.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """Get the firmware version of the board."""
+        return self._backend.firmware_version
+
+    def make_safe(self) -> None:
+        """Make this board safe."""
+        for servo in self._servos:
+            # Brake both motors.
```
Comment (kierdavis) [inline j5/boards/sr/v4/servo_board.py]:
```suggestion
            # Cut power to all servos.
```

File: tests/backends/console/sr/v4/test_servo_board.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def test_backend_set_servo_pos_out_of_range() -> None:
+    """Test that we throw an error if the value is out of range."""
+    backend = SRV4ServoBoardConsoleBackend(
+        "TestBoard",
+        console_class=MockConsole,
+    )
+    with pytest.raises(ValueError):
+        backend._console.expects = "Setting servo 10 to -1.0."  # type: ignore
```
Comment (kierdavis) [inline tests/backends/console/sr/v4/test_servo_board.py]:
```suggestion
        backend._console.expects = "Setting servo 18 to -1.0."  # type: ignore
```

It seems this didn't break the tests so it probably doesn't matter, but it's good to be consistent :P

File: j5/backends/backend.py [hunk 3]
```diff
@@ -32,6 +32,10 @@ def __new__(mcs, name, bases, namespace, **kwargs):  # type:ignore
         if cls.__name__ == "Backend":
             return cls
 
+        # Check if this is an abstract Backend.
+        if len(cls.__bases__) == 1 and cls.__base__.__name__ != "Backend":
```
Comment (kierdavis) [inline j5/backends/backend.py]: What is the aim of this bit? I don't really understand what its doing, and it looks like a bit of a hack. Don't most (if not all) concrete `Backend`s have more than one base class anyway?

File: stubs/usb/core.pyi [hunk 4]
```diff
@@ -4,10 +4,13 @@ Type stubs for usb.core
 Note that stubs are only written for the parts that we use.
 """
 
-from typing import Generator, Optional, Union
+from typing import Generator, Optional, Tuple, Union
 
 
 class Device:
+
+    _langids: Tuple[int, ...]
```
Comment (kierdavis) [inline stubs/usb/core.pyi]: Is this used?

File: j5/backends/backend.py (hunk 3 above)
Comment (trickeydan) [reply j5/backends/backend.py]: This identifies abstract Backends such as `RawUSBHardwareBackend`, which do not have a specific board associated with them and exempts them from the check.

Should no longer be necessary when #149 is done.

File: stubs/usb/core.pyi (hunk 4 above)
Comment (trickeydan) [reply stubs/usb/core.pyi]: Ah this was something I was using for debugging https://github.com/j5api/sr-servo-v4-fw/issues/5

Shouldn't be here.

**Comment by trickeydan** [comment]
> If you're working on a hardware backend, I think it would be best suited to a separate PR (huge PRs are not too enjoyable to manage/review).

Ok. I'll refactor the hardware out into a separate PR.

**Comment by trickeydan** [comment]
Removed Hardware implementation @kierdavis Ready for review.

File: j5/backends/backend.py (hunk 3 above)
Comment (trickeydan) [reply j5/backends/backend.py]: -> Hardware PR

**Pull request merged**